### PR TITLE
remove avx options from CUDA_NVCC_FLAGS on non-Intel systems

### PR DESCRIPTION
**Pull request by doichanj**
### Summary
There are still avx2 options when using CUDA backend

### Details and comments
I added branch to select options for CUDA_NVCC_FLAGS when using CUDA backend of Thrust simulator.

Also, I added to set AER_THRUST_SUPPORTED to TRUE is AER_THRUST_BACKEND is defined.

### CODE REVIEW

File: src/simulators/statevector/qubitvector_thrust.hpp [hunk 1]
```diff
@@ -799,8 +799,8 @@ void QubitVectorChunkContainer<data_t>::SetupP2P(int nDev)
       for(i=0;i<nDev;i++){
         if(i != m_iDevice){
           cudaDeviceCanAccessPeer(&m_p2pEnable[i],m_iDevice,i);
-          if(m_p2pEnable[i])
-            cudaDeviceEnablePeerAccess(i,0);
+//          if(m_p2pEnable[i])
+//            cudaDeviceEnablePeerAccess(i,0);
```
Comment (vvilpas) [inline src/simulators/statevector/qubitvector_thrust.hpp:803]: Is this change intended for this PR or was something done for testing?

File: CMakeLists.txt [hunk 2]
```diff
@@ -207,6 +207,8 @@ if(NOT DEFINED AER_THRUST_BACKEND)
     if(DEFINED AER_THRUST_BACKEND)
         set(AER_THRUST_SUPPORTED TRUE)
     endif()
+else()
+	set(AER_THRUST_SUPPORTED TRUE)
```
Comment (vvilpas) [inline CMakeLists.txt:211]: Is this change intended for this PR or was something done for testing?

Comment (vvilpas) [reply CMakeLists.txt:211]: Ok, I have just read the description of the PR. ;)

File: src/simulators/statevector/qubitvector_thrust.hpp (hunk 1 above)
Comment (doichanj) [reply src/simulators/statevector/qubitvector_thrust.hpp:803]: I found this will cause failure on unittest, so commented out
